config/application.php

```diff
@@ -25,35 +24,40 @@
  */
 $webroot_dir = $root_dir . '/web';
 
+/** A couple extra tweaks to help things run well on Pantheon. **/
+if (isset($_SERVER['HTTP_HOST'])) {
+    // HTTP is still the default scheme for now.
+    $scheme = 'http';
+    // If we have detected that the end use is HTTPS, make sure we pass that
+    // through here, so <img> tags and the like don't generate mixed-mode
+    // content warnings.
+    if (isset($_SERVER['HTTP_USER_AGENT_HTTPS']) && $_SERVER['HTTP_USER_AGENT_HTTPS'] == 'ON') {
+        $scheme = 'https';
+    }
+
+    /**
+     * URLs
+     */
+    Config::define('WP_HOME', $scheme . '://' . $_SERVER['HTTP_HOST']);
+    Config::define('WP_SITEURL', $scheme . '://' . $_SERVER['HTTP_HOST'] . '/wp');
+}
+
 /**
- * Use Dotenv to set required environment variables and load .env file in root
- * .env.local will override .env if it exists
+ * Use Dotenv to set required environment variables and load .env.pantheon file in root
+ * .env.local will override .env.pantheon if it exists
  */
 $env_files = file_exists($root_dir . '/.env.local')
-    ? ['.env', '.env.local']
-    : ['.env'];
+    ? ['.env.pantheon', '.env.local']
+    : ['.env.pantheon'];
 
 $dotenv = Dotenv\Dotenv::createUnsafeImmutable($root_dir, $env_files, false);
-if (file_exists($root_dir . '/.env')) {
+if (file_exists($root_dir . '/.env.pantheon')) {
     $dotenv->load();
-    $dotenv->required(['WP_HOME', 'WP_SITEURL']);
     if (!env('DATABASE_URL')) {
         $dotenv->required(['DB_NAME', 'DB_USER', 'DB_PASSWORD']);
     }
 }
 
-/**
- * Set up our global environment constant and load its config first
- * Default: production
- */
-define('WP_ENV', env('WP_ENV') ?: 'production');
```

Comment:
Is this constant used by anything? Should we define it based on PANTHEON_ENVIRONMENT?